### ISSUE TITLE
Pursuing learning from last saved model

### ISSUE THREAD

**Issue by benoit-combes** [MIC-DKFZ/nnUNet#238]
Dear developers,

Thanks a lot for this work. I would like to know if there is a practical procedure to rerun a Unet training from the last fitted parameters (using the same dataset). To be more precise, I recently run a Unet fitting whose error seems to still decrease notably after the standard 1000 epochs and I would like to "pursue" the training to see how long the decreasing is (without rerunning the first 1000 epochs). Could you comment on that ?

thanks a lot for your input,
Benoit

**Comment by FabianIsensee** [comment]
Hi,
this will be difficult to achieve, but is certainly not impossible. You need to create a separate trainer class that has max_num_epochs=2000 or however long you wish to train. Then you nee to make sure to load the old checkpoint before you resume training with this. This will require you to do some copying of files.
Due to the way the lr scheduling works this will most likely not improve your results significantly. lr almost linearly decreases over the course of the training, so in the last epoch it is almost 0. If you now restart from epoch 1000 and have a max of 2000, the lr will be approx 0.005 which is a lot higher then what the last checkpoint finished with, causing in instantaneous drop in accuracy.
If you want longer training I would not resume a previous checkpoint and instead create a trainer class with the appropriate number of epochs and then start from scratch.
But again, due to the nature of lr scheduling it is expected to see a decrease in fitting error towards the end. This is normal and does not indicate that the model is not done training.
Best,
Fabian
